docker-compose.yml

```diff
@@ -26,6 +29,6 @@ services:
       - .:/velox:delegated
     command: &cpp-command >
```

Comment:
what do you think about adding a sensible default here (8?) like:
```
    environment:
      NUM_THREADS: 8
```
This will allow to be overriden as you pointed out `docker-compose run -e NUM_THREADS=<NUMBER_OF_THREADS_TO_USE>`, in this case we can use the command as:
```
/bin/bash -c "
        make &&
        cd _build/release &&
        ctest -j$${NUM_THREADS} -VV --output-on-failure"
```
with the benefit that you also have control about the number of threads being used to run ctest.

Reply:
Hi @raulcd,

Did you actually mean 
```
/bin/bash -c "
        make  NUM_THREADS=$${NUM_THREADS} &&
        cd _build/release &&
        ctest -j$${NUM_THREADS} -VV --output-on-failure"
```
didn't you?

Anyway, yes this is a good idea: my only problem is that the Makefile override the value of NUM_THREADS only if its value is empty (NUM_THREADS is indeed an environmental variable added in the Dockerfile). I was thinking of a different solution, namely define a shell script incorporating the command above. The script can give us more freedom in choosing different default values for compilation and testing.
So, something like

```
services:
  ubuntu-cpp:
    image: facebookincubator/velox:amd64-ubuntu-22.04-cpp
    build:
      context: .
      dockerfile: scripts/ubuntu-22.04-cpp.dockerfile
    volumes:
      - .:/velox:delegated
   command: scripts/command.sh # or entrypoint
```
where in scripts/command.sh I put the commands above, providing default values (for testing) here.

What do you think?

Reply:
I didn't, as the environment variable is always defined it would already be able to be read on the container, we don't require to pass it when invoking the make command. I've tested locally to validate :)
I am ok creating a command or entrypoint script too as this could allow us to add more functionality in the future if needed.

Reply:
Sorry, you are right, I made confusion with the command make -jp. 

Ok then, let's use a script. I am going to add the new commit in the next days.

Thanks for your help!

Reply:
About this: `my only problem is that the Makefile override the value of NUM_THREADS only if its value is empty` I think it's fine to always define a value on docker-compose (either the default of 8 I was proposing) or the overriden one that you can use on CLI `-e NUM_THREADS=XX`. Makefile can use the total number of processors on environments where docker is not being used.

Reply:
Hi @raulcd,

I updated the PR with your indications. However, CircleCI is complaining about a "missed testing parallelization", but I think it's just because the entry command is written a different way now (bash script). 

Please let me know if it is alright or I made any mistake.

Thank you!